modules/php/Game/Utils.php

```diff
@@ -111,4 +115,48 @@ public static function hasMostKeepers($active_player_id)
     }
     return true;
   }
+
+  public static function playerHasNotYetPartiedInTurn()
+  {
+    // Party bonus can only be scored once by the same player in one turn.
+    return 0 == Utils::getGame()->getGameStateValue("playerTurnUsedPartyBonus");
+  }
+
+  public static function playerHasNotYetBeenPoorInTurn()
+  {
+    // Poor bonus can only be scored once by the same player in one turn.
+    return 0 == Utils::getGame()->getGameStateValue("playerTurnUsedPoorBonus");
+  }
+
+  public static function recheckForPartyBonus($player_id)
```

Comment:
I feel this could be called `check...` instead of `recheck...` but I'm being picky

Reply:
Well, we actually need to re-check for the poor and party bonus after every keeper card or action card played.
Hency my reasoning to call these functions recheck...
But you are right it is not consistent, because I did name the main function "checkBonusConditions", so I will adapt these also